### PR TITLE
[wpmlcore-5172] Created WPML_VC_Website_Builder_Translatable_Nodes

### PR DESCRIPTION
**Pull request by adrianosferreira**

### CODE REVIEW

File: src/class-wpml-vc-website-builder-translatable-nodes.php [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+	 * @var string
+	 */
+	private $type;
+
+	/** @var array */
+	private $nodes_to_translate;
+
+	public function __construct() {
+		$this->settings_field = self::SETTINGS_FIELD;
+		$this->type           = 'tag';
```
Comment (ghost) [inline src/class-wpml-vc-website-builder-translatable-nodes.php]: @adrianosferreira why not use 'tag' as const instead of var?

**Review by ghost — APPROVED** [review]
@adrianosferreira looks good only one minor comment

File: src/class-wpml-vc-website-builder-translatable-nodes.php (hunk 1 above)
Comment (adrianosferreira) [reply src/class-wpml-vc-website-builder-translatable-nodes.php]: @grey-rsi I'm following the pattern we used in Beaver Builder and Elementor integration modules.

Comment (ghost) [reply src/class-wpml-vc-website-builder-translatable-nodes.php]: fine for me then =)

**Comment by andreasciamanna** [comment]
I did my review, but I suggest @BruceOnTheGo to always do be one of the reviewers (at least for now), as he's one with better knowledge (than me, at least) of how Page Builders integration should work.

File: src/class-wpml-vc-website-builder-translatable-nodes.php [hunk 2]
```diff
@@ -0,0 +1,162 @@
+<?php
+
+/**
+ * Class WPML_VC_Website_Builder_Translatable_Nodes
+ */
+class WPML_VC_Website_Builder_Translatable_Nodes implements IWPML_Page_Builders_Translatable_Nodes {
```
Comment (brucepearson) [inline src/class-wpml-vc-website-builder-translatable-nodes.php]: @adrianosferreira This file in nearly identical to `WPML_Beaver_Builder_Translatable_Nodes`. I see that $settings is an array here and it's an object in BB. Can't we reuse code somehow for this?

The only real difference is the `initialize_nodes_to_translate` function

File: src/class-wpml-vc-website-builder-translatable-nodes.php [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+			$this->initialize_nodes_to_translate();
+		}
+
+		$strings = array();
+
+		foreach ( $this->nodes_to_translate as $node_type => $node_data ) {
+			if ( $this->conditions_ok( $node_data, $settings ) ) {
+				foreach ( $node_data['fields'] as $field ) {
+					$field_key = $field['field'];
+					if ( isset( $settings[ $field_key ] ) && trim( $settings[ $field_key ] ) ) {
```
Comment (brucepearson) [inline src/class-wpml-vc-website-builder-translatable-nodes.php]: @adrianosferreira If all this code is in a common abstract base class then you can call `$this->get_setting( $settings, $field_key )` here. Then in the child class you add 
```
protected function get_setting( $setting, $feild_key ) {
   return $setting[ $field_key ];
}
```

And in BB

```
protected function get_setting( $setting, $feild_key ) {
   return $setting->$field_key;
}
```

File: src/class-wpml-vc-website-builder-translatable-nodes.php (hunk 2 above)
Comment (adrianosferreira) [reply src/class-wpml-vc-website-builder-translatable-nodes.php]: @brucepearson basically the only difference is this one you said, object and array. I will look into it.

File: src/class-wpml-vc-website-builder-translatable-nodes.php (hunk 3 above)
Comment (adrianosferreira) [reply src/class-wpml-vc-website-builder-translatable-nodes.php]: @brucepearson makes sense.

Comment (adrianosferreira) [reply src/class-wpml-vc-website-builder-translatable-nodes.php]: @brucepearson please check the new commit here and also this new PR: https://github.com/OnTheGoSystems/wpml-page-builders/pull/3

I've done what you asked but didn't change Elementor and PB classes yet.

**Comment by adrianosferreira** [comment]
@brucepearson I've removed interface implementation.

**Pull request merged**